### PR TITLE
Add ruff

### LINKED ISSUE #603: Add support for ruff
https://github.com/charliermarsh/ruff

cc @charliermarsh since he likely would like to know about this too

### PR DESCRIPTION
**Pull request by pickfire**
Fix #603

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,6 +46,7 @@ _**Note:** The behavior of actions like this one is currently limited in the con
   - [Mypy](https://mypy.readthedocs.io/)
   - [oitnb](https://pypi.org/project/oitnb/)
   - [Pylint](https://pylint.pycqa.org)
+  - [Ruff](https://ruff.rs)
```
Comment (pickfire) [inline README.md:49]: @charliermarsh is it safe to assume https://ruff.rs to be the website? Currently it redirects to the github repository.

Comment (charliermarsh) [reply README.md]: :D Thanks for supporting Ruff! I haven't read through the code here but let me know if you have any questions.

File: test/linters/params/ruff.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+			stdout: "",
+		},
+		// Expected output of the parsing function
+		lintResult: {
+			isSuccess: false,
+			warning: [],
+			error: [],
+		},
+	};
+}
```
[the remark this answers is not in the document]
Comment (charliermarsh) [reply test/linters/params/ruff.js:65]: Haven't looked at this PR closely, but FYI we recently changed `--fix-only` to always return a code of 0, unless you provide `--exit-non-zero-on-fix`, in which case, it'll exit 1 if it fixes any violations.

Relevant link to the breaking change list: https://github.com/charliermarsh/ruff/blob/main/BREAKING_CHANGES.md#--fix-only-now-exits-with-a-zero-exit-code-unless---exit-non-zero-on-fix-is-specified-4146

**Comment by pickfire** [comment]
I don't get why is black linter failing in the tests.

**Comment by fvalverd** [comment]
> I don't get why is black linter failing in the tests.

`mypy` and `black` latest version are failing because the output format changed, I provided you a PR with the necessary changes to make them pass in here https://github.com/pickfire/lint-action/pull/1/files

File: test/linters/params/black.js [hunk 3]
```diff
@@ -9,38 +12,40 @@ const extensions = ["py"];
...
-	const stdoutFile1 = `--- file1.py	${TEST_DATE}\n+++ file1.py	${TEST_DATE}\n@@ -1,10 +1,10 @@\n var_1 = "hello"\n var_2 = "world"\n \n \n-def main ():  # Whitespace error\n+def main():  # Whitespace error\n     print("hello " + var_2)\n \n \n def add(num_1, num_2):\n     return num_1 + num_2\n@@ -19,8 +19,9 @@\n \n \n def divide(num_1, num_2):\n     return num_1 / num_2\n \n+\n # Blank lines error\n \n main()`;
-	const stdoutFile2 = `--- file2.py	${TEST_DATE}\n+++ file2.py	${TEST_DATE}\n@@ -1,2 +1,2 @@\n def add(num_1, num_2):\n-  return num_1 + num_2  # Indentation error\n+    return num_1 + num_2  # Indentation error`;
+	const absolutePathFile1 = join(dir, 'file1.py');
+	const stdoutFile1 = `--- ${absolutePathFile1}	${TEST_DATE}\n+++ ${absolutePathFile1}	${TEST_DATE}\n@@ -1,10 +1,10 @@\n var_1 = "hello"\n var_2 = "world"\n \n \n-def main ():  # Whitespace error\n+def main():  # Whitespace error\n     print("hello " + var_2)\n \n \n def add(num_1, num_2):\n     return num_1 + num_2\n@@ -19,8 +19,9 @@\n \n \n def divide(num_1, num_2):\n     return num_1 / num_2\n \n+\n # Blank lines error\n \n main()`;
+	const absolutePathFile2 = join(dir, 'file2.py');
```
Comment (fvalverd) [inline test/linters/params/black.js:17]: Sorry, I forgot to check the format before doing the PR, here is a quick fix for the linter error :+1: 

```suggestion
	const absolutePathFile1 = join(dir, "file1.py");
	const stdoutFile1 = `--- ${absolutePathFile1}	${TEST_DATE}\n+++ ${absolutePathFile1}	${TEST_DATE}\n@@ -1,10 +1,10 @@\n var_1 = "hello"\n var_2 = "world"\n \n \n-def main ():  # Whitespace error\n+def main():  # Whitespace error\n     print("hello " + var_2)\n \n \n def add(num_1, num_2):\n     return num_1 + num_2\n@@ -19,8 +19,9 @@\n \n \n def divide(num_1, num_2):\n     return num_1 / num_2\n \n+\n # Blank lines error\n \n main()`;
	const absolutePathFile2 = join(dir, "file2.py");
```

**Comment by tylerlaprade** [comment]
@fvalverd, tests still seem to be failing

**Comment by fvalverd** [comment]
There are 3 failures:
* [macos](https://github.com/wearerequired/lint-action/actions/runs/6451254167/job/17511638029?pr=620): that is `swift`, nothing to do with this PR or Python in general
* [ubuntu](https://github.com/wearerequired/lint-action/actions/runs/6451254167/job/17511638126?pr=620): that is also `swift`, it has nothing to do with this PR or Python in general
* [windows](https://github.com/wearerequired/lint-action/actions/runs/6451254167/job/17511638195?pr=620): those are related to `ruff` and `black`, probably the syntax of the output is the problem, but I don't have access to that OS to corroborate the problem

**Comment by luzfcb** [comment]
I don't know if it makes anything easier, but ruff currently supports some output formats, such as `github` and `json`

https://docs.astral.sh/ruff/settings/#output-format

```bash
$ ruff check --output-format github .
::error title=Ruff (C408),file=/home/luzfcb/projects/labcodes/pythondotorg/events/tests/test_utils.py,line=36,col=17,endLine=36,endColumn=60::events/tests/test_utils.py:36:17: C408 Unnecessary `dict` call (rewrite as a literal)
```


```bash
$ ruff check --output-format json .
[
  {
    "cell": null,
    "code": "C408",
    "end_location": {
      "column": 60,
      "row": 36
    },
    "filename": "/home/luzfcb/projects/labcodes/pythondotorg/events/tests/test_utils.py",
    "fix": null,
    "location": {
      "column": 17,
      "row": 36
    },
    "message": "Unnecessary `dict` call (rewrite as a literal)",
    "noqa_row": 36,
    "url": "https://docs.astral.sh/ruff/rules/unnecessary-collection-call"
  },
........
```

**Comment by BroFromSpace** [comment]
Any progress on this PR?

**Comment by pickfire** [comment]
@BroFromSpace I no longer need this, I guess you can continue working on it. Will close this.

**Pull request closed without merging**